### PR TITLE
Series of small commits, fixes motivated by WSL (Windows Subsystem for Linux)

### PR DESCRIPTION
**Pull request by gc00**
The eight commit messages tell what are the changes.  This will be easy to review if you simply review each of the eight commits, one-by-one.

* After reviewing my commits, please do not push it in yourself, unless you are willing to take ownership of my code and be responsible for any bugs in my code.  There are several subtleties here, and I would like the privilege of reflecting on any review comments before pushing this in, rather than later committing fixes for earlier commits.

* Also, if you do decide to push this in yourself, please don't push it in as a single merge.  I know that github makes that easier to do.  But it means that 'git log' shows eight semi-unrelated commits as a single giant merge.  :-(

* WSL is not yet working.  At checkpoint time, doLocking gets stuck on fcntl (maybe because the underlying Windows filesystem is different??), but WSL should now be close to working.

### CODE REVIEW

File: src/mtcp/restore_libc.c [hunk 1]
```diff
@@ -503,6 +551,7 @@ TLSInfo_VerifyPidTid(pid_t pid, pid_t tid)
 {
   pid_t tls_pid, tls_tid;
   char *addr = (char *)get_tls_base_addr();
+  addr = (char *)get_tls_base_addr();
```
Comment (rohgarg) [inline src/mtcp/restore_libc.c]: Why are we doing this again? Isn't the line above (line no. 553) doing the same assignment?

File: configure.ac [hunk 2]
```diff
@@ -401,6 +405,18 @@ else
   AC_SUBST([AARCH64_HOST], [no])
 fi
 
+AC_MSG_CHECKING([if this is WSL (Windows Subystem for Linux)])
+if uname -r | grep Microsoft > /dev/null; then
+  is_wsl='yes'
+  wsl_version=`uname -r`
+else
+  is_wsi='no'
```
Comment (rohgarg) [inline configure.ac]: Typo: `is_wsi` --> `is_wsl`

File: src/mtcp/restore_libc.c (hunk 1 above)
Comment (gc00) [reply src/mtcp/restore_libc.c]: Thanks.  This must have been a typo.

File: configure.ac (hunk 2 above)
Comment (gc00) [reply configure.ac]: Done.

**Comment by gc00** [comment]
@rohgarg, Thanks.  I've addressed the comments, and am ready for a re-review.

**Comment by gc00** [comment]
@rohgarg, ping :-) (ready for re-review)

File: Makefile.in [hunk 3]
```diff
@@ -106,7 +106,9 @@ _hooksdir=.git/hooks
 add-git-hooks: ${_hooksdir}/pre-commit ${_hooksdir}/post-rewrite
 
 ${_hooksdir}/%: util/hooks/%
-	cd ${_hooksdir} && ln -s ../../$< .
+	if test -d ${_hooksdir}; then \
```
Comment (karya0) [inline Makefile.in]: Is `.git/hooks` missing on WSL?

**Pull request merged**